File: modules/core/src/encoder/BooleanEncoder.ts
import {Encoder} from '.';

export class BooleanEncoder implements Encoder<boolean> {

  public encode(value: boolean): string {

    return +value + '';

  }

  public decode(value: string): boolean {

    // less operations than !!parseInt(value, 2)
    return value === '1';

  }

}

File: GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/encoder/field/BooleanEncoder.java
package com.gdpr.phase2.encoder.field;

import com.gdpr.phase2.encoder.BaseEncoder;

public class BooleanEncoder implements BaseEncoder<Boolean> {
//    public final String encode(Boolean value) {
//        return value + "";
//    }

    private BooleanEncoder() {
    }

    private static final BooleanEncoder instance = new BooleanEncoder();
    public static BooleanEncoder getInstance() {
        return instance;
    }
    public final Boolean decode(String value){
        return value.equals("1");
    }
}
